src/middlewared/middlewared/etc_files/krb5.keytab.py

```diff
@@ -1,40 +1,57 @@
 import logging
 import os
 import base64
-
-from middlewared.utils import run
+import stat
+import subprocess
+from contextlib import suppress
 
 logger = logging.getLogger(__name__)
 kdir = "/etc/kerberos"
 keytabfile = "/etc/krb5.keytab"
 ktutil_cmd = "/usr/sbin/ktutil"
 
 
-async def write_keytab(db_keytabname, db_keytabfile):
-    temp_keytab = f'{kdir}/{db_keytabname}'
-    if not os.path.exists(kdir):
-        os.mkdir(kdir)
-    if os.path.exists(temp_keytab):
-        os.remove(temp_keytab)
-    with open(temp_keytab, "wb") as f:
-        f.write(db_keytabfile)
+def set_mode(fd, mode):
+    if stat.S_IMODE(os.fstat(fd).st_mode) != mode:
+        os.fchmod(fd, mode)
+
+
+def write_keytab(db_keytabname, db_keytabfile):
+    def opener(path, flags):
+        return os.open(path, flags, dir_fd=dirfd, mode=0o600)
+
+    with suppress(FileExistsError):
+        os.mkdir(kdir, mode=0o700)
+
+    try:
+        dirfd = os.open(kdir, os.O_DIRECTORY)
+        set_mode(dirfd, 0o600)
```

Comment:
Question: In the `22.12` code base, I see you verifying that the directory perms are `0o700`, where as here, we're blindly setting the directory perms to `0o600`. Is this a FreeBSD specific change ? What's the story here ?

Reply:
Typo.